### PR TITLE
APIs to set and remove path segments.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 1]
```diff
@@ -692,6 +696,38 @@ public Builder addEncodedPathSegment(String encodedPathSegment) {
       return this;
     }
 
+    public Builder setPathSegment(int index, String pathSegment) {
+      if (pathSegment == null) throw new IllegalArgumentException("pathSegment == null");
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Missing index bounds check. Or if you are falling back to List to do the work then tests are missing.

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 2]
```diff
@@ -993,33 +1029,36 @@ private void resolvePath(String input, int pos, int limit) {
...
         encodedPathSegments.add(segment);
       }
-
       if (addTrailingSlash) {
         encodedPathSegments.add("");
       }
     }
 
+    private boolean isDot(String input) {
+      return input.equalsIgnoreCase(".") || input.equalsIgnoreCase("%2e");
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Ignore case on dot?

Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Seems intentional and I'm curious.

**Comment by JakeWharton** [comment]
LGTM!

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java (hunk 1 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: I'll add a test!

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java (hunk 2 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Seemed simpler to keep all cases symmetric. Probably unnecessary. I eyeballed it both ways.

I'll change it.

**Pull request merged**